src/server/generic_family.cc

```diff
@@ -398,10 +399,7 @@ OpStatus Renamer::UpdateDest(Transaction* t, EngineShard* es) {
       if (src_res_.ref_val.ObjType() == OBJ_STRING) {
         pv_.SetString(str_val_);
       }
-      auto add_res =
-          db_slice.AddNew(t->GetDbContext(), dest_key, std::move(pv_), src_res_.expire_ts);
-      dest_it = add_res.it;
```

Comment:
I belive you should update dest_it

Reply:
but `dest_it` is now a reference to `res.it`

Reply:
(also - we can't update references to point to another object)

Reply:
ops you are right